From 94111293fb044909dff9153049fa7b1d52fb1748 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Mon, 8 Feb 2016 17:02:32 -0800
Subject: [PATCH] laird: wb45n: Move hw_init_hook to driver file

driver/hw_init_hook.c is where the custom hw_init_hook used to be and should
be. So fix up and prep for the WB50's custom hw_init by moving it to a common
driver file.

Bug: 8628
---
 board/at91sam9x5ek/at91sam9x5ek.c | 15 +--------------
 driver/hw_init_hook.c             | 23 ++++++++++++++++++-----
 2 files changed, 19 insertions(+), 19 deletions(-)

diff --git a/board/at91sam9x5ek/at91sam9x5ek.c b/board/at91sam9x5ek/at91sam9x5ek.c
index 745b672..a026b92 100644
--- a/board/at91sam9x5ek/at91sam9x5ek.c
+++ b/board/at91sam9x5ek/at91sam9x5ek.c
@@ -46,20 +46,7 @@
 #include "board_hw_info.h"
 
 #ifdef CONFIG_USER_HW_INIT
-static void hw_init_hook(void)
-{
-	/* Setup WB45NBT custom GPIOs */
-	const struct pio_desc gpios[] = {
-	/*  {"NAME",        PIN NUMBER,     VALUE, ATTRIBUTES, TYPE },*/
-		{"CHIP_PWD_L",	AT91C_PIN_PA(28),	0, PIO_DEFAULT, PIO_OUTPUT},
-		{"VBUS_SENSE",	AT91C_PIN_PB(11),	0, PIO_DEFAULT, PIO_INPUT},
-		{"VBUS_EN",	    AT91C_PIN_PB(12),	0, PIO_DEFAULT, PIO_OUTPUT},
-		{"IRQ",	        AT91C_PIN_PB(18),	0, PIO_PULLUP,  PIO_INPUT},
-		{(char *)0,	0, 0, PIO_DEFAULT, PIO_PERIPH_A},
-	};
-
-	pio_configure(gpios);
-}
+extern void hw_init_hook(void);
 #endif
 
 static void at91_dbgu_hw_init(void)
diff --git a/driver/hw_init_hook.c b/driver/hw_init_hook.c
index 03dee85..315d400 100644
--- a/driver/hw_init_hook.c
+++ b/driver/hw_init_hook.c
@@ -1,8 +1,21 @@
+#include "arch/at91_pio.h"
+#include "gpio.h"
+
 void hw_init_hook(void)
 {
-    /*
-     * This is a placeholder for the users custom init routine
-     * Overwrite this file with your own routine to init
-     * your own H/W features.
-     */
+#if defined (CONFIG_AT91SAM9X5EK)
+	/* Setup WB45NBT custom GPIOs */
+	const struct pio_desc gpios[] = {
+	/*  {"NAME",        PIN NUMBER,     VALUE, ATTRIBUTES, TYPE },*/
+		{"CHIP_PWD_L",	AT91C_PIN_PA(28),	0, PIO_DEFAULT, PIO_OUTPUT},
+		{"VBUS_SENSE",	AT91C_PIN_PB(11),	0, PIO_DEFAULT, PIO_INPUT},
+		{"VBUS_EN",	    AT91C_PIN_PB(12),	0, PIO_DEFAULT, PIO_OUTPUT},
+		{"IRQ",	        AT91C_PIN_PB(18),	0, PIO_PULLUP,  PIO_INPUT},
+		{(char *)0,	0, 0, PIO_DEFAULT, PIO_PERIPH_A},
+	};
+#else
+#error CONFIG_USER_HW_INIT needs to be defined for your board.
+#endif
+
+	pio_configure(gpios);
 }
-- 
1.9.1

